From cae18ef8f2795e5f584fe450b9619de176e72e02 Mon Sep 17 00:00:00 2001
From: random-nick <random-nick@email.com>
Date: Fri, 24 Jul 2020 00:04:19 +0200
Subject: [PATCH] [youtube] Fix age gate detection

---
 youtube_dl/extractor/youtube.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/youtube_dl/extractor/youtube.py b/youtube_dl/extractor/youtube.py
index c27f2cd9523e..685b0cd64157 100644
--- a/youtube_dl/extractor/youtube.py
+++ b/youtube_dl/extractor/youtube.py
@@ -1825,7 +1825,7 @@ def extract_player_response(player_response, video_id):
         # Get video info
         video_info = {}
         embed_webpage = None
-        if re.search(r'player-age-gate-content">', video_webpage) is not None:
+        if self._html_search_meta('og:restrictions:age', video_webpage, default=None) == "18+":
             age_gate = True
             # We simulate the access to the video from www.youtube.com/v/{video_id}
             # this can be viewed without login into Youtube
