diff --git a/youtube_dl/extractor/bandcamp.py b/youtube_dl/extractor/bandcamp.py
index f14b407dc82c..ad1812320663 100644
--- a/youtube_dl/extractor/bandcamp.py
+++ b/youtube_dl/extractor/bandcamp.py
@@ -91,10 +91,11 @@ def _real_extract(self, url):
         duration = None
 
         formats = []
-        track_info = self._parse_json(
-            self._search_regex(
-                r'trackinfo\s*:\s*\[\s*({.+?})\s*\]\s*,\s*?\n',
-                webpage, 'track info', default='{}'), title)
+        trackinfo_block = self._search_regex(
+            r'trackinfo&quot;:\[\s*({.+?})\s*\],&quot;',
+            webpage, 'track info', default='{}')
+        quoted_json = trackinfo_block.replace('&quot;', '"')
+        track_info = self._parse_json(quoted_json, title)
         if track_info:
             file_ = track_info.get('file')
             if isinstance(file_, dict):
@@ -117,7 +118,7 @@ def _real_extract(self, url):
 
         def extract(key):
             return self._search_regex(
-                r'\b%s\s*["\']?\s*:\s*(["\'])(?P<value>(?:(?!\1).)+)\1' % key,
+                r',&quot;%s&quot;:(&quot;)(?P<value>(?:(?!&quot;).)+)&quot;' % key,
                 webpage, key, default=None, group='value')
 
         artist = extract('artist')
