From 9b2231c13dae2705369f3abdfaa19c32d79ef6a4 Mon Sep 17 00:00:00 2001
From: ethus3h <kolubat@gmail.com>
Date: Sun, 23 Apr 2017 02:28:42 -0400
Subject: [PATCH] Add option for configuring resize border thickness

---
 plugins/resize.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/resize.c b/plugins/resize.c
index d5ce472..439ae92 100644
--- a/plugins/resize.c
+++ b/plugins/resize.c
@@ -1331,6 +1331,7 @@ resizePaintRectangle (CompScreen              *s,
     CompTransform sTransform = *transform;
 
     RESIZE_SCREEN (s);
+    RESIZE_DISPLAY (s->display);
 
     resizeGetPaintRectangle (s->display, &box);
 
