diff --git a/src/libaudcore/tuple.cc b/src/libaudcore/tuple.cc
index 603a4752e..4444f3a21 100644
--- a/src/libaudcore/tuple.cc
+++ b/src/libaudcore/tuple.cc
@@ -819,8 +819,8 @@ EXPORT void Tuple::generate_fallbacks ()
         {
             if (second && ! artist && ! album)
             {
-                data->set_str (FallbackArtist, second);
-                data->set_str (FallbackAlbum, first);
+                data->set_str (FallbackArtist, first);
+                data->set_str (FallbackAlbum, nullptr);
             }
             else
                 data->set_str (artist ? FallbackAlbum : FallbackArtist, first);
