diff --git a/audacious.desktop b/audacious.desktop
index 804b27ce3..9ce805ddf 100644
--- a/audacious.desktop
+++ b/audacious.desktop
@@ -6,7 +6,7 @@ GenericName=Music Player
 Comment=Listen to music
 Icon=audacious
 Categories=AudioVideo;Audio;Player;
-Exec=audacious %U
+Exec=bash -c 'audacious %U; sleep 1; systemctl --user restart mpris-scrobbler.service'
 TryExec=audacious
 StartupNotify=false
 Terminal=false
