diff --git a/metadata/ezoom.xml.in b/metadata/ezoom.xml.in
index bcafcc9..f634d46 100644
--- a/metadata/ezoom.xml.in
+++ b/metadata/ezoom.xml.in
@@ -40,6 +40,7 @@
 		<option type="key" name="zoom_out_key">
 		    <_short>Zoom Out</_short>
 		    <_long>Zoom Out</_long>
+		    <default>&lt;Shift&gt;&lt;Primary&gt;&lt;Alt&gt;u</default>
 		</option>
 		<option type="button" name="zoom_box_button">
 		    <_short>Zoom Box</_short>
@@ -138,7 +139,7 @@
 		    <option type="key" name="fit_to_window_key">
 			<_short>Fit zoomed area to window</_short>
 			<_long>Zooms in/out so the focused window is zoomed to the maximum while still being fully visible.</_long>
-			<default>&lt;Super&gt;f</default>
+			<default>&lt;Shift&gt;&lt;Primary&gt;&lt;Alt&gt;z</default>
 		    </option>
 		</subgroup>
 	    </group>
@@ -167,7 +168,7 @@
 		<option type="bool" name="sync_mouse">
 		    <_short>Sync Mouse</_short>
 		    <_long>Always keep the zoomed area in sync with where the mouse is, and vice versa. Use this if you don't intend to draw a scaled pointer or hide the original pointer. The zoomed area will move as you move your mouse.</_long>
-		    <default>true</default>
+		    <default>false</default>
 		</option>
 		<option type="bool" name="scale_mouse">
 		    <_short>Scale the mouse pointer</_short>
@@ -200,7 +201,7 @@
 		<option type="bool" name="mouse_pan">
 		    <_short>Mouse panning</_short>
 		    <_long>Pan the zoom area when the mouse cursor moves outside the visible area.</_long>
-		    <default>false</default>
+		    <default>true</default>
 		</option>
 		<option type="int" name="restrain_margin">
 		    <_short>Mouse Restrain Margin</_short>
