diff --git a/metadata/winrules.xml.in b/metadata/winrules.xml.in
index 6abaf49..2e200a6 100644
--- a/metadata/winrules.xml.in
+++ b/metadata/winrules.xml.in
@@ -11,17 +11,17 @@
 				<option name="skiptaskbar_match" type="match">
 					<_short>Skip taskbar</_short>
 					<_long>Don't show application in taskbar</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="skippager_match" type="match">
 					<_short>Skip pager</_short>
 					<_long>Don't show application in pager</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="above_match" type="match">
 				<_short>Above</_short>
 					<_long>Above others windows</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="below_match" type="match">
 					<_short>Below</_short>
@@ -31,7 +31,7 @@
 				<option name="sticky_match" type="match">
 					<_short>Sticky</_short>
 					<_long>Sticky windows</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="fullscreen_match" type="match">
 					<_short>Fullscreen</_short>
@@ -41,7 +41,7 @@
 				<option name="maximize_match" type="match">
 					<_short>Maximized</_short>
 					<_long>Maximized windows</_long>
-				    <default/>
+					<default>class=Xarchiver</default>
 				</option>
 				<option name="no_argb_match" type="match">
 					<_short>No ARGB visuals</_short>
@@ -51,12 +51,12 @@
 				<option name="no_move_match" type="match">
 					<_short>Non movable windows</_short>
 					<_long>Set window as non movable</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="no_resize_match" type="match">
 					<_short>Non resizable windows</_short>
 					<_long>Set window as non resizable</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="no_minimize_match" type="match">
 					<_short>Non minimizable windows</_short>
@@ -66,7 +66,7 @@
 				<option name="no_maximize_match" type="match">
 					<_short>Non maximizable windows</_short>
 					<_long>Set window as non maximizable</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 				<option name="no_close_match" type="match">
 					<_short>Non closable windows</_short>
@@ -76,7 +76,7 @@
 				<option name="no_focus_match" type="match">
 					<_short>No focus</_short>
 					<_long>Windows will not have focus</_long>
-					<default/>
+					<default>class=Onboard</default>
 				</option>
 			</group>
 			<group>
