diff -uraN a/colorfilter.xml.in b/colorfilter.xml.in
--- a/colorfilter.xml.in	2020-03-11 20:51:43.000000000 -0400
+++ b/colorfilter.xml.in	2020-03-11 20:52:56.000000000 -0400
@@ -18,7 +18,7 @@
       <option name="toggle_screen_key" type="key">
         <_short>Toggle screen filtering</_short>
         <_long>Toggle filtering for every window on the screen.</_long>
-        <default>&lt;Super&gt;&lt;Alt&gt;d</default>
+        <default>&lt;Shift&gt;&lt;Alt&gt;f</default>
       </option>
       <option name="switch_filter_key" type="key">
         <_short>Switch filter</_short>
@@ -30,7 +30,7 @@
       <option name="filter_decorations" type="bool">
         <_short>Filter window decorations</_short>
         <_long>Window decorations (borders and titlebar) will be filtered as well as window contents if enabled.</_long>
-        <default>false</default>
+        <default>true</default>
       </option>
       <option name="filter_match" type="match">
         <_short>Filtered windows</_short>
@@ -40,7 +40,7 @@
       <option name="exclude_match" type="match">
         <_short>Exclude windows</_short>
         <_long>Windows to exclude from filtering.</_long>
-        <default>Dock | type=Desktop</default>
+        <default></default>
       </option>
       <option name="notifications_enable" type="bool">
         <_short>Enable notifications upon filter change</_short>
@@ -62,25 +62,7 @@
         <_long>List of plain text filter files that should be applied when filtering windows or screen.</_long>
         <type>string</type>
         <default>
-	    <value>blackandwhite</value>
-	    <value>blueish-filter</value>
-	    <value>colornegative</value>
-	    <value>contrast</value>
-	    <value>deuteranopia</value>
-	    <value>grayscale</value>
-	    <value>negative</value>
-	    <value>negative-green</value>
-	    <value>negative-lightness</value>
 	    <value>nightmode</value>
-	    <value>nightmodeb</value>
-	    <value>nightmodec</value>
-	    <value>nightmoded</value>
-	    <value>protanopia</value>
-	    <value>sepia</value>
-	    <value>swap-green-blue</value>
-	    <value>swap-red-blue</value>
-	    <value>swap-red-green</value>
-	    <value>swap-hue</value>
 	</default>
         <hints>file;</hints>
       </option>
