diff --git a/metadata/neg.xml.in b/metadata/neg.xml.in
index fa7e9ed..0756174 100644
--- a/metadata/neg.xml.in
+++ b/metadata/neg.xml.in
@@ -17,21 +17,46 @@
 			</option>
 			<option name="screen_toggle_key" type="key">
 				<_short>Toggle Screen Negative</_short>
-				<_long>Toggle screen negative</_long>
+				<_long>Toggle all windows negative, except for windows in the &quot;Screen Exclusions&quot; list</_long>
 				<default>&lt;Super&gt;m</default>
 			</option>
+			<option name="matched_toggle_key" type="key">
+				<_short>Toggle Matched Windows Negative</_short>
+				<_long>Toggles the windows negative matched by the &quot;Negative Windows&quot; list</_long>
+				<default></default>
+			</option>
 		</display>
 		<screen>
 			<option name="toggle_by_default" type="bool">
 				<_short>Auto-Toggle Matched Windows</_short>
-				<_long>Automatically toggle windows in the match list by default</_long>
-				<default>false</default>
+				<_long>Automatically toggle windows in the &quot;Matched Windows&quot; list by default</_long>
+				<default>false</default><!-- This default must be kept in sync with the initialization of ns->matchNeg in neg.c. -->
 			</option>
 			<option name="neg_match" type="match">
-				<_short>Negative Windows</_short>
-				<_long>Windows to affect when negating</_long>
+				<_short>Matched Windows</_short>
+				<_long>Windows to affect when using the &quot;Toggle Matched Windows Negative&quot; keybinding or &quot;Auto-Toggle Matched Windows&quot;</_long>
 				<default>!(type=Desktop)</default>
 			</option>
+			<option name="toggle_screen_by_default" type="bool">
+				<_short>Auto-Toggle Screen</_short>
+				<_long>Automatically toggle all windows by default, except for those in the &quot;Screen Exclusions&quot; list</_long>
+				<default>false</default><!-- This default must be kept in sync with the initialization of ns->isNeg in neg.c. -->
+			</option>
+			<option name="exclude_match" type="match">
+				<_short>Screen Exclusions</_short>
+				<_long>Windows to exclude when toggling the screen.</_long>
+				<default>type=Desktop | type=Dock</default>
+			</option>
+			<option name="preserve_toggled" type="bool">
+				<_short>Preserve Toggled Windows</_short>
+				<_long>When windows have been toggled using the &quot;Toggle Windows Negative&quot; keybinding, preserve their state when toggling the screen. Note that toggled windows' state will not be preserved between Compiz restarts.</_long>
+				<default>false</default>
+			</option>
+			<option name="clear_toggled" type="bool">
+				<_short>Auto-Clear Toggled Window State</_short>
+				<_long>A window that has been toggled using the &quot;Toggle Windows Negative&quot; keybinding will have its state discarded when toggling a match group that includes that window.</_long>
+				<default>true</default>
+			</option>
 		</screen>
 	</plugin>
 </compiz>
diff --git a/src/neg/neg.c b/src/neg/neg.c
index 1ed10c0..abbd629 100644
--- a/src/neg/neg.c
+++ b/src/neg/neg.c
@@ -46,7 +46,14 @@ typedef struct _NEGScreen
 
     DrawWindowTextureProc drawWindowTexture;
 
-    Bool isNeg; /* negative screen flag */
+    Bool isNeg; /* negative screen flag: controlled by "Auto-Toggle Screen"
+                   checkbox */
+    Bool keyNegToggled; /* screen is toggled using the "Toggle Screen Negative"
+                           keybinding */
+    Bool matchNeg; /* match group is toggled: controlled by "Auto-Toggle
+                      Matched Windows" checkbox */
+    Bool keyMatchToggled; /* match group is toggled using the "Toggle Matched
+                             Windows Negative" keybinding */
 
     int negFunction;
     int negAlphaFunction;
@@ -54,8 +61,15 @@ typedef struct _NEGScreen
 
 typedef struct _NEGWindow
 {
-    Bool isNeg; /* negative window flag */
-    Bool matched;
+    Bool isNeg; /* negative window flag: controlled by NEGUpdateState function */
+    Bool keyNegToggled; /* window has been individually toggled using the
+                           "Toggle Window Negative" keybinding (will be unset
+                           when Preserve Toggled Windows means the window
+                           should be using its previous state) */
+    Bool keyNegPreserved; /* window has been individually toggled using the
+                           "Toggle Window Negative" keybinding. This preserves
+                           the window state between screen toggles for Preserve
+                           Toggled Windows. */
 } NEGWindow;
 
 #define GET_NEG_CORE(c) \
@@ -81,32 +95,113 @@ typedef struct _NEGWindow
 static void
 NEGUpdateState (CompWindow *w)
 {
+    NEG_SCREEN (w->screen);
     NEG_WINDOW (w);
 
-    /* check include list */
-    if (matchEval (negGetNegMatch (w->screen), w))
-		nw->isNeg = !nw->isNeg;
-	else
-		nw->isNeg = FALSE;
+    Bool windowState;
+
+    /* Decide whether the given window should be negative or not, depending on
+       the various parameters that can affect this, and set windowState thus */
+
+	windowState = FALSE;
+
+    /* Whole screen toggle state */
+    if (! matchEval (negGetExcludeMatch (w->screen), w)) {
+	if (ns->isNeg)
+	    windowState = !windowState;
+	if (ns->keyNegToggled)
+	    windowState = !windowState;
+    }
+
+    /* Matched set toggle state */
+    if (matchEval (negGetNegMatch (w->screen), w)) {
+	if (ns->matchNeg)
+	    windowState = !windowState;
+	if (ns->keyMatchToggled)
+	    windowState = !windowState;
+    }
+
+    /* Individual window state */
+    if (nw->keyNegToggled)
+	windowState = !windowState;
+
+    /* Now that we know what this window's state should be, push the value to
+       its nw->isNeg. */
+    nw->isNeg = windowState;
 
     /* cause repainting */
     addWindowDamage (w);
 }
 
+static void
+NEGUpdateScreen (CompScreen *s)
+{
+    CompWindow *w;
+
+    /* update every window */
+    for (w = s->windows; w; w = w->next)
+	if (w)
+	    NEGUpdateState (w);
+}
+
+static void
+NEGWindowUpdateKeyToggle (CompWindow *w)
+{
+    NEG_WINDOW (w);
+
+    if (!negGetPreserveToggled (w->screen))
+	return;
+
+    if (nw->keyNegToggled)
+	nw->keyNegToggled = FALSE;
+    else if (nw->keyNegPreserved)
+	nw->keyNegToggled = TRUE;
+}
+
+static void
+NEGToggleWindow (CompWindow *w)
+{
+    NEG_WINDOW (w);
+
+    nw->keyNegToggled = !nw->keyNegToggled;
+    nw->keyNegPreserved = !nw->keyNegPreserved;
+
+    /* cause repainting */
+    NEGUpdateState (w);
+}
+
 static void
 NEGToggleScreen (CompScreen *s)
 {
+    NEG_SCREEN (s);
     CompWindow *w;
 
-    NEG_SCREEN(s);
+    /* update toggle state for relevant windows */
+    for (w = s->windows; w; w = w->next)
+	if (w && negGetPreserveToggled (s) && ! matchEval (negGetExcludeMatch (s), w))
+	    NEGWindowUpdateKeyToggle (w);
 
     /* toggle screen negative flag */
-    ns->isNeg = !ns->isNeg;
+    ns->keyNegToggled = !ns->keyNegToggled;
+
+    NEGUpdateScreen (s);
+}
+
+static void
+NEGToggleMatches (CompScreen *s)
+{
+    NEG_SCREEN (s);
+    CompWindow *w;
 
-    /* toggle every window */
+    /* update toggle state for relevant windows */
     for (w = s->windows; w; w = w->next)
-	if (w)
-	    NEGUpdateState (w);
+	if (w && negGetPreserveToggled (s) && matchEval (negGetNegMatch (s), w))
+	    NEGWindowUpdateKeyToggle (w);
+
+    /* toggle match negative flag */
+    ns->keyMatchToggled = !ns->keyMatchToggled;
+
+    NEGUpdateScreen (s);
 }
 
 static Bool
@@ -123,7 +218,7 @@ negToggle (CompDisplay     *d,
     w = findWindowAtDisplay (d, xid);
 
     if (w)
-	NEGUpdateState (w);
+	NEGToggleWindow (w);
 
     return TRUE;
 }
@@ -147,6 +242,25 @@ negToggleAll (CompDisplay     *d,
     return TRUE;
 }
 
+static Bool
+negToggleMatched (CompDisplay     *d,
+	      CompAction      *action,
+	      CompActionState state,
+	      CompOption      *option,
+	      int             nOption)
+{
+    CompScreen *s;
+    Window     xid;
+
+    xid = getIntOptionNamed (option, nOption, "root", 0);
+    s = findScreenAtDisplay (d, xid);
+
+    if (s)
+	NEGToggleMatches (s);
+
+    return TRUE;
+}
+
 static int
 getNegFragmentFunction (CompScreen  *s,
 			CompTexture *texture,
@@ -549,15 +663,8 @@ static void
 NEGWindowAdd (CompScreen *s,
 	      CompWindow *w)
 {
-	NEG_SCREEN (s);
-	NEG_WINDOW (w);
-
-	nw->matched = matchEval (negGetNegMatch (s), w);
-
-	/* nw->isNeg is initialized to FALSE in InitWindow, so we only
-	have to toggle it to TRUE if necessary */
-	if (ns->isNeg && nw->matched)
-		NEGUpdateState (w);
+	/* Run matching logic on the new window */
+	NEGUpdateState (w);
 }
 
 static void
@@ -568,48 +675,60 @@ NEGScreenOptionChanged (CompScreen       *s,
     switch (num)
     {
     case NegScreenOptionToggleByDefault:
-    {
-		CompWindow *w;
-
-		NEG_SCREEN (s);
-
-		ns->isNeg = opt[NegScreenOptionToggleByDefault].value.b;
+	{
+	    NEG_SCREEN (s);
+	    CompWindow *w;
 
+	    if (negGetClearToggled (s)) {
 		for (w = s->windows; w; w = w->next)
 		{
+		    if (matchEval (negGetNegMatch (w->screen), w)) {
 			NEG_WINDOW (w);
-			if (ns->isNeg)
-			{
-				if (!nw->isNeg)
-					NEGUpdateState (w);
-			}
-			else
-			{
-				if (nw->isNeg)
-					NEGUpdateState (w);
-			}
+			nw->keyNegToggled = FALSE;
+			nw->keyNegPreserved = FALSE;
+		    }
 		}
+	    }
+
+	    ns->matchNeg = negGetToggleByDefault (s);
+
+	    NEGUpdateScreen (s);
 	}
-    break;
+	break;
     case NegScreenOptionNegMatch:
 	{
-	    CompWindow *w;
+	    NEGUpdateScreen (s);
+	}
+	break;
+    case NegScreenOptionToggleScreenByDefault:
+	{
 	    NEG_SCREEN (s);
+	    CompWindow *w;
 
-	    for (w = s->windows; w; w = w->next)
-	    {
+	    if (negGetClearToggled (s)) {
+		for (w = s->windows; w; w = w->next)
+		{
+		    if (! matchEval (negGetExcludeMatch (w->screen), w)) {
 			NEG_WINDOW (w);
+			nw->keyNegToggled = FALSE;
+			nw->keyNegPreserved = FALSE;
+		    }
+		}
+	    }
 
-			nw->matched = matchEval (negGetNegMatch (w->screen), w);
+	    ns->isNeg = negGetToggleScreenByDefault (s);
 
-			if (nw->matched)
-			{
-				if ((ns->isNeg || negGetToggleByDefault (s)) && !nw->isNeg)
-					NEGUpdateState (w);
-			}
-			else if (nw->isNeg)
-				NEGUpdateState (w);
-	    }
+	    NEGUpdateScreen (s);
+	}
+	break;
+    case NegScreenOptionExcludeMatch:
+	{
+	    NEGUpdateScreen (s);
+	}
+	break;
+    case NegScreenOptionPreserveToggled:
+	{
+	    NEGUpdateScreen (s);
 	}
 	break;
     default:
@@ -694,8 +813,9 @@ NEGInitDisplay (CompPlugin  *p,
 	return FALSE;
     }
 
-    negSetWindowToggleKeyInitiate (d, negToggle);
-    negSetScreenToggleKeyInitiate (d, negToggleAll);
+    negSetWindowToggleKeyInitiate  (d, negToggle);
+    negSetScreenToggleKeyInitiate  (d, negToggleAll);
+    negSetMatchedToggleKeyInitiate (d, negToggleMatched);
 
     d->base.privates[displayPrivateIndex].ptr = nd;
 
@@ -735,13 +855,23 @@ NEGInitScreen (CompPlugin *p,
     /* initialize the screen variables
      * you know what happens if you don't
      */
-    ns->isNeg = FALSE;
+    ns->isNeg           = FALSE; /* Keep in sync with default
+                                    toggle_screen_by_default value from
+                                    neg.xml.in */
+    ns->keyNegToggled   = FALSE;
+    ns->matchNeg        = FALSE; /* Keep in sync with default toggle_by_default
+                                    value from neg.xml.in */
+    ns->keyMatchToggled = FALSE;
 
     ns->negFunction      = 0;
     ns->negAlphaFunction = 0;
 
     negSetToggleByDefaultNotify (s, NEGScreenOptionChanged);
     negSetNegMatchNotify (s, NEGScreenOptionChanged);
+    negSetToggleScreenByDefaultNotify (s, NEGScreenOptionChanged);
+    negSetExcludeMatchNotify (s, NEGScreenOptionChanged);
+    negSetPreserveToggledNotify (s, NEGScreenOptionChanged);
+    negSetClearToggledNotify (s, NEGScreenOptionChanged);
 
     /* wrap overloaded functions */
     WRAP (ns, s, drawWindowTexture, NEGDrawWindowTexture);
@@ -781,8 +911,9 @@ NEGInitWindow (CompPlugin *p,
     if (!nw)
 	return FALSE;
 
-    nw->isNeg       = FALSE;
-    nw->matched     = FALSE;
+    nw->isNeg           = FALSE;
+    nw->keyNegToggled   = FALSE;
+    nw->keyNegPreserved = FALSE;
 
     w->base.privates[ns->windowPrivateIndex].ptr = nw;
 
@@ -858,4 +989,3 @@ getCompPluginInfo(void)
 {
     return &NEGVTable;
 }
-
