diff --git a/metadata/neg.xml.in b/metadata/neg.xml.in
index ea5e71e..72090cc 100644
--- a/metadata/neg.xml.in
+++ b/metadata/neg.xml.in
@@ -50,12 +50,12 @@
 			<option name="preserve_toggled" type="bool">
 				<_short>Preserve Toggled Windows</_short>
 				<_long>When windows have been toggled using the &quot;Toggle Windows Negative&quot; keybinding, preserve their state when toggling the screen. Note that toggled windows' state will not be preserved between Compiz restarts. To use this option, disable &quot;Auto-Clear Toggled Window State&quot;.</_long>
-				<default>false</default>
+				<default>true</default>
 			</option>
 			<option name="clear_toggled" type="bool">
 				<_short>Auto-Clear Toggled Window State</_short>
 				<_long>A window that has been toggled using the &quot;Toggle Windows Negative&quot; keybinding will have its state discarded when toggling a match group that includes that window. This will clear window states even if &quot;Preserve Toggled Windows&quot; is selected.</_long>
-				<default>true</default>
+				<default>false</default>
 			</option>
 		</screen>
 	</plugin>
