diff --git a/metadata/neg.xml.in b/metadata/neg.xml.in
index fa7e9ed..5ad6526 100644
--- a/metadata/neg.xml.in
+++ b/metadata/neg.xml.in
@@ -22,6 +22,11 @@
 			</option>
 		</display>
 		<screen>
+			<option name="default_negative" type="bool">
+				<_short>Negative by default</_short>
+				<_long>Default screen state is negative</_long>
+				<default>false</default>
+			</option>
 			<option name="toggle_by_default" type="bool">
 				<_short>Auto-Toggle Matched Windows</_short>
 				<_long>Automatically toggle windows in the match list by default</_long>
diff --git a/src/neg/neg.c b/src/neg/neg.c
index 45b1721..be9a309 100644
--- a/src/neg/neg.c
+++ b/src/neg/neg.c
@@ -564,6 +564,11 @@ NEGScreenOptionChanged (CompScreen       *s,
 {
     switch (num)
     {
+    case NegScreenOptionDefaultNegative:
+    {
+		NEGToggleScreen (s);
+    }
+    break;
     case NegScreenOptionToggleByDefault:
     {
 		CompWindow *w;
@@ -738,6 +743,7 @@ NEGInitScreen (CompPlugin *p,
     ns->negFunction      = 0;
     ns->negAlphaFunction = 0;
 
+    negSetDefaultNegativeNotify (s, NEGScreenOptionChanged);
     negSetToggleByDefaultNotify (s, NEGScreenOptionChanged);
     negSetNegMatchNotify (s, NEGScreenOptionChanged);
 
@@ -856,4 +862,3 @@ getCompPluginInfo(void)
 {
     return &NEGVTable;
 }
-
