diff --git a/src/scrobbler2/scrobbler.h b/src/scrobbler2/scrobbler.h
index 9236a3159..07c96a27f 100644
--- a/src/scrobbler2/scrobbler.h
+++ b/src/scrobbler2/scrobbler.h
@@ -24,7 +24,7 @@
 
 #define SCROBBLER_API_KEY "4b4f73bda181868353f9b438604adf52"
 #define SCROBBLER_SHARED_SECRET "716cc0a784bb62835de5bd674e65eb57"
-#define SCROBBLER_URL "https://ws.audioscrobbler.com/2.0/"
+#define SCROBBLER_URL "https://listenbrainz.org/2.0/"
 
 
 extern const PluginPreferences configuration;
