diff --git a/src/scrobbler2/scrobbler.cc b/src/scrobbler2/scrobbler.cc
index 9c44b9465..45a0593c2 100644
--- a/src/scrobbler2/scrobbler.cc
+++ b/src/scrobbler2/scrobbler.cc
@@ -7,6 +7,7 @@
  * It is licensed under the GNU General Public License, version 3.
  */
 
+#include <regex>
 #include <glib/gstdio.h>
 
 //audacious includes
@@ -91,8 +92,12 @@ static gboolean queue_track_to_scrobble (void * data) {
 
     char *queuepath = g_strconcat(aud_get_path(AudPath::UserDir),"/scrobbler.log", nullptr);
 
+    std::regex filter ("-(([-\\w]{11})|(\\d{9,10}))$");
+
     StringBuf artist = clean_string (playing_track.get_str (Tuple::Artist));
-    StringBuf title  = clean_string (playing_track.get_str (Tuple::Title));
+    StringBuf titletemp  = clean_string (playing_track.get_str (Tuple::Title));
+    char * titletempb =  std::regex_replace((const char *)titletemp, filter, "").c_str();
+    StringBuf title = str_copy(titletempb);
     StringBuf album  = clean_string (playing_track.get_str (Tuple::Album));
 
     int track  = playing_track.get_int (Tuple::Track);
diff --git a/src/scrobbler2/scrobbler_communication.cc b/src/scrobbler2/scrobbler_communication.cc
index b3d343c37..8795dbdd0 100644
--- a/src/scrobbler2/scrobbler_communication.cc
+++ b/src/scrobbler2/scrobbler_communication.cc
@@ -3,6 +3,7 @@
 #include <stdarg.h>
 #include <stdlib.h>
 #include <sys/time.h>
+#include <regex>
 #include <curl/curl.h>
 
 #include <glib.h>
@@ -500,8 +501,12 @@ static void send_now_playing() {
    */
   Tuple curr_track = now_playing_track.ref ();
 
+  std::regex filter ("-(([-\\w]{11})|(\\d{9,10}))$");
+
   StringBuf artist = clean_string (curr_track.get_str (Tuple::Artist));
-  StringBuf title = clean_string (curr_track.get_str (Tuple::Title));
+  StringBuf titletemp = clean_string (curr_track.get_str (Tuple::Title));
+  char * titletempb =  std::regex_replace((const char *)titletemp, filter, "").c_str();
+  StringBuf title = str_copy(titletempb);
   StringBuf album = clean_string (curr_track.get_str (Tuple::Album));
 
   int track  = curr_track.get_int (Tuple::Track);
@@ -517,7 +522,8 @@ static void send_now_playing() {
      "duration", (const char *) length_str, "api_key", SCROBBLER_API_KEY,
      "sk", (const char *) session_key);
 
-    gboolean success = send_message_to_lastfm(playingmsg);
+    // gboolean success = send_message_to_lastfm(playingmsg);
+    gboolean success = true;
 
     if (success == false) {
       AUDDBG("Network problems. Could not send \"now playing\" to last.fm\n");
