diff --git a/ccm/Settings.py b/ccm/Settings.py
index 380e988..ae7eba5 100644
--- a/ccm/Settings.py
+++ b/ccm/Settings.py
@@ -295,17 +295,38 @@ class FamilyStringSetting(StockSetting):
             previewWidget.modify_font(fd)
 
 class FileStringSetting(StringSetting):
+    FILE = 0
+    DIRECTORY = 1
+    PATH = 2
 
-    def __init__(self, setting, List=False, isImage=False, isDirectory=False):
+    def __init__(self, setting, List=False, isImage=False, pickMode=0):
         self.isImage = isImage
-        self.isDirectory = isDirectory
+        self.pickMode = pickMode
         StringSetting.__init__(self, setting, List=List)
 
     def _Init(self):
         StringSetting._Init(self)
-        self.FileButton = FileButton(self.Setting.Plugin.Context, self.Entry,
-            self.isDirectory, self.isImage)
-        self.Box.pack_start(self.FileButton, False, False, 0)
+        if self.pickMode == FileStringSetting.PATH:
+            self.FileButton = FileButton(
+                self.Setting.Plugin.Context,
+                self.Entry,
+                False, self.isImage
+            )
+            self.DirButton = FileButton(
+                self.Setting.Plugin.Context,
+                self.Entry,
+                True, False
+            )
+            self.Box.pack_start(self.FileButton, False, False, 0)
+            self.Box.pack_start(self.DirButton, False, False, 0)
+        else:
+            self.FileButton = FileButton(
+                self.Setting.Plugin.Context,
+                self.Entry,
+                self.pickMode == FileStringSetting.DIRECTORY,
+                self.isImage
+            )
+            self.Box.pack_start(self.FileButton, False, False, 0)
 
 class EnumSetting(StockSetting):
 
@@ -1624,14 +1645,26 @@ def MakeStringSetting (setting, List=False):
 
     if setting.Hints:
         if "file" in setting.Hints:
-            if "image" in setting.Hints:
-                return FileStringSetting (setting, isImage=True, List=List)
-            else:
-                return FileStringSetting (setting, List=List)
+            return FileStringSetting (
+                setting,
+                isImage="image" in setting.Hints,
+                List=List
+            )
         elif "family" in setting.Hints:
             return FamilyStringSetting (setting)
         elif "directory" in setting.Hints:
-            return FileStringSetting (setting, isDirectory=True, List=List)
+            return FileStringSetting (
+                setting,
+                pickMode=FileStringSetting.DIRECTORY,
+                List=List
+            )
+        elif "path" in setting.Hints:
+            return FileStringSetting(
+                setting,
+                pickMode=FileStringSetting.PATH,
+                isImage="image" in setting.Hints,
+                List=List
+            )
         else:
             return StringSetting (setting, List=List)
     elif (List and setting.Info[1][2]) or \
diff --git a/ccm/Widgets.py b/ccm/Widgets.py
index 7bfa206..31ef312 100644
--- a/ccm/Widgets.py
+++ b/ccm/Widgets.py
@@ -1633,8 +1633,12 @@ class FileButton (Gtk.Button):
         self._path = path
 
         self.set_tooltip_text(_("Browse..."))
-        self.set_image(Gtk.Image.new_from_icon_name("document-open",
-                                                    Gtk.IconSize.BUTTON))
+        if self._directory:
+            self.set_image(Gtk.Image.new_from_icon_name(
+                "text-x-generic", Gtk.IconSize.BUTTON))
+        else:
+            self.set_image(Gtk.Image.new_from_icon_name(
+                "document-open", Gtk.IconSize.BUTTON))
         self.connect('clicked', self.open_dialog)
 
     def set_path (self, value):
