diff --git a/metadata/dialog.xml.in b/metadata/dialog.xml.in
index 50ade3d..43aa023 100644
--- a/metadata/dialog.xml.in
+++ b/metadata/dialog.xml.in
@@ -25,7 +25,7 @@
 				<option name="opacity" type="float">
 					<_short>Opacity</_short>
 					<_long>The opacity of windows with open dialogs</_long>
-					<default>80</default>
+					<default>95</default>
 					<min>0</min>
 					<max>100</max>
 					<precision>1</precision>
@@ -33,7 +33,7 @@
 				<option name="saturation" type="float">
 					<_short>Saturation</_short>
 					<_long>The saturation of windows with open dialogs</_long>
-					<default>18</default>
+					<default>100</default>
 					<min>0</min>
 					<max>100</max>
 					<precision>1</precision>
@@ -41,7 +41,7 @@
 				<option name="brightness" type="float">
 					<_short>Brightness</_short>
 					<_long>The brightness of windows with open dialogs</_long>
-					<default>60</default>
+					<default>85</default>
 					<min>0</min>
 					<max>100</max>
 					<precision>1</precision>
