diff --git a/metadata/scalefilter.xml.in b/metadata/scalefilter.xml.in
index 1902103..0d38bdd 100644
--- a/metadata/scalefilter.xml.in
+++ b/metadata/scalefilter.xml.in
@@ -41,33 +41,33 @@
 			<_short>Font family</_short>
 			<_long>Name of the font to display title.</_long>
 			<hints>family;</hints>
-			<default>Sans</default>
+			<default>Wreathe System</default>
 		</option>
 		<option name="font_bold" type="bool">
 		    <_short>Bold Font</_short>
 		    <_long>Use bold font for window title display</_long>
-		    <default>true</default>
+		    <default>false</default>
 		</option>
 		<option name="font_size" type="int">
 		    <_short>Font Size</_short>
 		    <_long>Font size for window title display</_long>
 		    <min>6</min>
 		    <max>144</max>
-		    <default>24</default>
+		    <default>48</default>
 		</option>
 		<option name="border_size" type="int">
 		    <_short>Title Border Size</_short>
 		    <_long>Size of the border around the window title</_long>
 		    <min>1</min>
 		    <max>60</max>
-		    <default>5</default>
+		    <default>8</default>
 		</option>
 		<option name="font_color" type="color">
 		    <_short>Font Color</_short>
 		    <_long>Font color of the window title</_long>
 		    <default>
-			<red>0xffff</red>
-			<green>0xffff</green>
+			<red>0xf0f0</red>
+			<green>0xb0b0</green>
 			<blue>0xffff</blue>
 			<alpha>0xffff</alpha>
 		    </default>
@@ -76,9 +76,9 @@
 		    <_short>Background Color</_short>
 		    <_long>Background color of the window title</_long>
 		    <default>
-			<red>0x0000</red>
+			<red>0x1414</red>
 			<green>0x0000</green>
-			<blue>0x0000</blue>
+			<blue>0x3737</blue>
 			<alpha>0x9999</alpha>
 		    </default>
 		</option>
