diff --git a/metadata/grid.xml.in b/metadata/grid.xml.in
index c8a1fcc..6e7ca2e 100644
--- a/metadata/grid.xml.in
+++ b/metadata/grid.xml.in
@@ -15,47 +15,47 @@
 		<option name="put_left_key" type="key">
 		    <_short>Put Left</_short>
 		    <_long>Move window to the left edge</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_4</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;q</default>
 		</option>
 		<option name="put_right_key" type="key">
 		    <_short>Put Right</_short>
 		    <_long>Move window to the right edge</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_6</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;e</default>
 		</option>
 		<option name="put_top_key" type="key">
 		    <_short>Put Top</_short>
 		    <_long>Move window to the top edge</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_8</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;2</default>
 		</option>
 		<option name="put_bottom_key" type="key">
 		    <_short>Put Bottom</_short>
 		    <_long>Move window to the bottom edge</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_2</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;s</default>
 		</option>
 		<option name="put_topleft_key" type="key">
 		    <_short>Put Top Left</_short>
 		    <_long>Move window to the top left corner</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_7</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;1</default>
 		</option>
 		<option name="put_topright_key" type="key">
 		    <_short>Put Top Right</_short>
 		    <_long>Move window to the top right corner</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_9</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;3</default>
 		</option>
 		<option name="put_bottomleft_key" type="key">
 		    <_short>Put Bottom Left</_short>
 		    <_long>Move window to the bottom left corner</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_1</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;a</default>
 		</option>
 		<option name="put_bottomright_key" type="key">
 		    <_short>Put Bottom Right</_short>
 		    <_long>Move window to the bottom right corner</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_3</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;d</default>
 		</option>
 		<option name="put_maximize_key" type="key">
 		    <_short>Maximize</_short>
 		    <_long>Maximize window</_long>
-		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_0</default>
+		    <default>&lt;Shift&gt;&lt;Alt&gt;w</default>
 		</option>
 		<option name="cycle_sizes" type="key">
 		    <_short>Cycle Through Multiple Sizes</_short>
@@ -69,7 +69,7 @@
 		    <_short>Resize Actions</_short>
 		    <_long>Window resize action</_long>
 		    <option name="top_left_corner_action" type="int">
-			<_short>Upper Left Corner</_short>
+			<_short>Top Left Corner</_short>
 			<_long>Action to be performed when window is dropped on the top left corner</_long>
 			<default>7</default>
 			<min>0</min>
@@ -171,7 +171,7 @@
 			</desc>
 		    </option>
 		    <option name="top_right_corner_action" type="int">
-			<_short>Upper Right Corner</_short>
+			<_short>Top Right Corner</_short>
 			<_long>Action to be performed when window is dropped on the top right corner</_long>
 			<default>9</default>
 			<min>0</min>
@@ -522,13 +522,21 @@
 		    <default>350</default>
 		    <min>0</min>
 		</option>
+                <option name="outline_thickness" type="float">
+                    <_short>Outline Thickness</_short>
+                    <_long>Outline Thickness</_long>
+                    <default>2.0</default>
+                    <min>1.0</min>
+                    <max>8.0</max>
+                    <precision>1.0</precision>
+                </option>
 		<option name="outline_color" type="color">
 		    <_short>Outline Color</_short>
 		    <_long>Color of the resize indicator outline</_long>
 		    <default>
-			<red>0x2f2f</red>
-			<green>0x2f2f</green>
-			<blue>0x4f4f</blue>
+			<red>0x6464</red>
+			<green>0x6464</green>
+			<blue>0x9797</blue>
 			<alpha>0x9f9f</alpha>
 		    </default>
 		</option>
@@ -536,10 +544,10 @@
 		    <_short>Fill Color</_short>
 		    <_long>Fill color of the resize indicator</_long>
 		    <default>
-			<red>0x2f2f</red>
-			<green>0x2f2f</green>
-			<blue>0x4f4f</blue>
-			<alpha>0x4f4f</alpha>
+			<red>0xe1e1</red>
+			<green>0xcdcd</green>
+			<blue>0xffff</blue>
+			<alpha>0xa9a9</alpha>
 		    </default>
 		</option>
 	    </group>
