diff --git a/internetarchive/cli/ia_move.py b/internetarchive/cli/ia_move.py
index 540235c..0531d15 100644
--- a/internetarchive/cli/ia_move.py
+++ b/internetarchive/cli/ia_move.py
@@ -32,7 +32,8 @@
 from __future__ import print_function, absolute_import
 import sys
 
-from docopt import docopt
+from docopt import docopt, printable_usage
+from schema import Schema, And, Use, SchemaError
 
 from internetarchive.cli import ia_copy
 from internetarchive.cli.argparser import get_args_dict
@@ -43,6 +44,22 @@ def main(argv, session):
     src_path = args['<src-identifier>/<src-file>']
     dest_path = args['<dest-identifier>/<dest-file>']
 
+        # Validate args.
+    s = Schema({
+        str: Use(bool),
+        '--metadata': list,
+        '--header': list,
+        '<src-identifier>/<src-file>': And(str, lambda x: '/' in x,
+            error='Source not formatted correctly. See usage example.'),
+        '<dest-identifier>/<dest-file>': And(str, lambda x: '/' in x,
+            error='Destiantion not formatted correctly. See usage example.'),
+    })
+    try:
+        args = s.validate(args)
+    except SchemaError as exc:
+        print('{0}\n{1}'.format(str(exc), printable_usage(__doc__)), file=sys.stderr)
+        sys.exit(1)
+
     headers = get_args_dict(args['--header'])
     # Add keep-old-version by default.
     if 'x-archive-keep-old-version' not in args['--header']:
diff --git a/internetarchive/item.py b/internetarchive/item.py
index a9547d8..075f263 100644
--- a/internetarchive/item.py
+++ b/internetarchive/item.py
@@ -772,7 +772,7 @@ def upload(self, files,
             total_files = recursive_file_count(files, item=self, checksum=False)
         for f in files:
             if (isinstance(f, string_types) and os.path.isdir(f)) \
-                    or (os.path.isdir(f[-1])):
+                    or (isinstance(f, tuple) and os.path.isdir(f[-1])):
                 if isinstance(f, tuple):
                     remote_dir_name = f[0].strip('/')
                     f = f[-1]
