diff --git a/src/main-process/atom-window.coffee b/src/main-process/atom-window.coffee
index 9bbdcfc2576..a30558d459b 100644
--- a/src/main-process/atom-window.coffee
+++ b/src/main-process/atom-window.coffee
@@ -156,16 +156,6 @@ class AtomWindow
       @atomApplication.removeWindow(this)
       @resolveClosedPromise()
 
-    @browserWindow.on 'unresponsive', =>
-      return if @isSpec
-
-      chosen = dialog.showMessageBox @browserWindow,
-        type: 'warning'
-        buttons: ['Force Close', 'Keep Waiting']
-        message: 'Editor is not responding'
-        detail: 'The editor is not responding. Would you like to force close it or just keep waiting?'
-      @browserWindow.destroy() if chosen is 0
-
     @browserWindow.webContents.on 'crashed', =>
       if @headless
         console.log "Renderer process crashed, exiting"
