diff --git a/metadata/blur.xml.in b/metadata/blur.xml.in
index d6b8ca91..53d47c08 100644
--- a/metadata/blur.xml.in
+++ b/metadata/blur.xml.in
@@ -72,7 +72,7 @@
 		<_long>Gaussian radius</_long>
 		<default>3</default>
 		<min>1</min>
-		<max>15</max>
+		<max>128</max>
 	    </option>
 	    <option name="gaussian_strength" type="float">
 		<_short>Gaussian Strength</_short>
diff --git a/plugins/blur.c b/plugins/blur.c
index 5d7303ad..a0cf6eef 100644
--- a/plugins/blur.c
+++ b/plugins/blur.c
@@ -35,7 +35,7 @@
 
 static CompMetadata blurMetadata;
 
-#define BLUR_GAUSSIAN_RADIUS_MAX 15
+#define BLUR_GAUSSIAN_RADIUS_MAX 128
 
 #define BLUR_FILTER_4X_BILINEAR 0
 #define BLUR_FILTER_GAUSSIAN    1
@@ -2876,7 +2876,7 @@ static const CompMetadataOptionInfo blurScreenOptionInfo[] = {
     { "alpha_blur_match", "match", 0, 0, 0 },
     { "alpha_blur", "bool", 0, 0, 0 },
     { "filter", "int", RESTOSTRING (0, BLUR_FILTER_LAST), 0, 0 },
-    { "gaussian_radius", "int", "<min>1</min><max>15</max>", 0, 0 },
+    { "gaussian_radius", "int", "<min>1</min><max>128</max>", 0, 0 },
     { "gaussian_strength", "float", "<min>0.0</min><max>1.0</max>", 0, 0 },
     { "mipmap_lod", "float", "<min>0.1</min><max>8.0</max>", 0, 0 },
     { "saturation", "int", "<min>0</min><max>100</max>", 0, 0 },
