diff --git a/metadata/annotate.xml.in b/metadata/annotate.xml.in
index 377563c4..321eda83 100644
--- a/metadata/annotate.xml.in
+++ b/metadata/annotate.xml.in
@@ -8,7 +8,7 @@
 	    <option name="initiate_free_draw_button" type="button">
 		<short>Initiate Free Draw</short>
 		<long>Initiate freehand drawing</long>
-		<default>&lt;Super&gt;&lt;Alt&gt;Button1</default>
+		<default>&lt;Shift&gt;&lt;Primary&gt;Button2</default>
 	    </option>
 	    <option name="initiate_line_button" type="button">
 		<short>Initiate Line</short>
@@ -33,12 +33,12 @@
 	    <option name="erase_button" type="button">
 		<short>Initiate erase</short>
 		<long>Initiate annotate erasing</long>
-		<default>&lt;Super&gt;&lt;Alt&gt;Button3</default>
+		<default>&lt;Shift&gt;&lt;Primary&gt;&lt;Alt&gt;Button2</default>
 	    </option>
 	    <option name="clear_key" type="key">
 		<short>Clear</short>
 		<long>Clear</long>
-		<default>&lt;Super&gt;&lt;Alt&gt;k</default>
+		<default>&lt;Shift&gt;&lt;Primary&gt;colon</default>
 	    </option>
 	    <option name="clear_button" type="button">
 		<short>Clear</short>
