diff --git a/metadata/fade.xml.in b/metadata/fade.xml.in
index 585e6a20..383b1f67 100644
--- a/metadata/fade.xml.in
+++ b/metadata/fade.xml.in
@@ -67,7 +67,7 @@
 	    <option name="minimize_open_close" type="bool">
 		<short>Fade On Minimize/Open/Close</short>
 		<long>Fade effect on minimize/open/close window events</long>
-		<default>true</default>
+		<default>false</default>
 	    </option>
 	    <option name="dim_unresponsive" type="bool">
 		<short>Dim Unresponsive Windows</short>
