diff --git a/metadata/scale.xml.in b/metadata/scale.xml.in
index 2dc9f149..b1111933 100644
--- a/metadata/scale.xml.in
+++ b/metadata/scale.xml.in
@@ -28,9 +28,7 @@
 		<option name="initiate_edge" type="edge">
 		    <_short>Initiate Window Picker</_short>
 		    <_long>Layout and start transforming windows</_long>
-		    <default>
-			<edge name="TopRight"/>
-		    </default>
+		    <default></default>
 		    <allowed edgednd="true"/>
 		</option>
 		<option name="initiate_key" type="key">
@@ -46,14 +44,19 @@
 		    <_short>Initiate Window Picker For All Windows</_short>
 		    <_long>Layout and start transforming all windows</_long>
 		    <allowed edgednd="true"/>
+		    <default>
+			<edge name="Bottom"/>
+		    </default>
 		</option>
 		<option name="initiate_all_button" type="button">
 		    <_short>Initiate Window Picker For All Windows</_short>
 		    <_long>Layout and start transforming all windows</_long>
+		    <default>&lt;LeftEdge&gt;&lt;BottomEdge&gt;&lt;BottomLeftEdge&gt;Button1</default>
 		</option>
 		<option name="initiate_all_key" type="key">
 		    <_short>Initiate Window Picker For All Windows</_short>
 		    <_long>Layout and start transforming all windows</_long>
+		    <default>F13</default>
 		</option>
 		<option name="initiate_group_edge" type="edge">
 		    <_short>Initiate Window Picker For Window Group</_short>
@@ -87,7 +90,7 @@
 		<option name="show_desktop" type="bool">
 		    <_short>Click Desktop to Show Desktop</_short>
 		    <_long>Enter Show Desktop mode when Desktop is clicked during Scale</_long>
-		    <default>true</default>
+		    <default>false</default>
 		</option>
 	    </group>
 	</display>
@@ -112,7 +115,7 @@
 		<option name="timestep" type="float">
 		    <_short>Timestep</_short>
 		    <_long>Scale timestep</_long>
-		    <default>1.2</default>
+		    <default>0.1</default>
 		    <min>0.1</min>
 		    <max>50</max>
 		    <precision>0.1</precision>
@@ -120,12 +123,12 @@
 		<option name="darken_back" type="bool">
 		    <_short>Darken Background</_short>
 		    <_long>Darken background when scaling windows</_long>
-		    <default>true</default>
+		    <default>false</default>
 		</option>
 		<option name="opacity" type="int">
 		    <_short>Opacity</_short>
 		    <_long>Amount of opacity in percent</_long>
-		    <default>75</default>
+		    <default>90</default>
 		    <min>0</min>
 		    <max>100</max>
 		</option>
