diff --git a/metadata/wall.xml.in b/metadata/wall.xml.in
index 0fa06c80..3c579f38 100644
--- a/metadata/wall.xml.in
+++ b/metadata/wall.xml.in
@@ -25,12 +25,12 @@
 				<option name="miniscreen" type="bool">
 					<short>Show Live Viewport Previews</short>
 					<long>Show live viewport previews in switcher window</long>
-					<default>false</default>
+					<default>true</default>
 				</option>
 				<option name="preview_timeout" type="float">
 					<short>Switch Target Preview Visibility Time</short>
 					<long>Duration (in s) the switch target preview should remain visible after switching ends.</long>
-					<default>0.4</default>
+					<default>0.129900</default>
 					<min>0.0</min>
 					<max>2.0</max>
 					<precision>0.05</precision>
@@ -45,14 +45,14 @@
 				<option name="edge_radius" type="int">
 					<short>Edge Radius</short>
 					<long>Radius of the rounded edge</long>
-					<default>5</default>
+					<default>0</default>
 					<min>0</min>
 					<max>20</max>
 				</option>
 				<option name="border_width" type="int">
 					<short>Border Width</short>
 					<long>Width of the border between the previews</long>
-					<default>10</default>
+					<default>8</default>
 					<min>0</min>
 					<max>30</max>
 				</option>
@@ -60,9 +60,9 @@
 					<short>Outline Color</short>
 					<long>Outline color of the switcher window.</long>
 					<default>
-						<red>0x3333</red>
-						<green>0x3333</green>
-						<blue>0x3333</blue>
+						<red>0xd0d0</red>
+						<green>0x8181</green>
+						<blue>0xffff</blue>
 						<alpha>0xd998</alpha>
 						</default>
 				</option>
@@ -73,7 +73,7 @@
 						<short>Base Color</short>
 						<long>First color for the background gradient of the switcher window.</long>
 						<default>
-							<red>0xcccc</red>
+							<red>0xe5e5</red>
 							<green>0xcccc</green>
 							<blue>0xe665</blue>
 							<alpha>0xd998</alpha>
@@ -83,9 +83,9 @@
 						<short>Highlight Color</short>
 						<long>Second color for the background gradient of the switcher window.</long>
 						<default>
-							<red>0xf332</red>
-							<green>0xf332</green>
-							<blue>0xffff</blue>
+							<red>0xdcdc</red>
+							<green>0xbcbc</green>
+							<blue>0xf7f7</blue>
 							<alpha>0xd998</alpha>
 						</default>
 					</option>
@@ -93,9 +93,9 @@
 						<short>Shadow Color</short>
 						<long>Third color for the background gradient of the switcher window.</long>
 						<default>
-							<red>0xf332</red>
-							<green>0xf332</green>
-							<blue>0xffff</blue>
+							<red>0xe5e5</red>
+							<green>0xcccc</green>
+							<blue>0xe6e6</blue>
 							<alpha>0xd998</alpha>
 						</default>
 					</option>
@@ -107,20 +107,20 @@
 						<short>Base Color</short>
 						<long>First color for the thumb gradient of the switcher window.</long>
 						<default>
-							<red>0x3333</red>
-							<green>0x3333</green>
-							<blue>0x3333</blue>
-							<alpha>0x5999</alpha>
+							<red>0xffff</red>
+							<green>0x0000</green>
+							<blue>0x0000</blue>
+							<alpha>0xffff</alpha>
 						</default>
 					</option>
 					<option name="thumb_gradient_highlight_color" type="color">
 						<short>Highlight Color</short>
 						<long>Second color for the thumb gradient of the switcher window.</long>
 						<default>
-							<red>0x3fff</red>
-							<green>0x3fff</green>
-							<blue>0x3fff</blue>
-							<alpha>0x3fff</alpha>
+							<red>0xffff</red>
+							<green>0x0000</green>
+							<blue>0x0000</blue>
+							<alpha>0xffff</alpha>
 						</default>
 					</option>
 				</subgroup>
@@ -132,8 +132,8 @@
 						<long>First color for the thumb highlight gradient of the switcher window.</long>
 						<default>
 							<red>0xffff</red>
-							<green>0xffff</green>
-							<blue>0xffff</blue>
+							<green>0x0000</green>
+							<blue>0x0000</blue>
 							<alpha>0xf332</alpha>
 						</default>
 					</option>
@@ -141,9 +141,9 @@
 						<short>Shadow Color</short>
 						<long>Second color for the thumb highlight gradient of the switcher window.</long>
 						<default>
-							<red>0xdfff</red>
-							<green>0xdfff</green>
-							<blue>0xdfff</blue>
+							<red>0xffff</red>
+							<green>0x0000</green>
+							<blue>0x0000</blue>
 							<alpha>0xa665</alpha>
 						</default>
 					</option>
@@ -155,9 +155,9 @@
 						<short>Base Color</short>
 						<long>First color of the arrow of the switcher window.</long>
 						<default>
-							<red>0xe665</red>
-							<green>0xe665</green>
-							<blue>0xe665</blue>
+							<red>0xb5b5</red>
+							<green>0x5656</green>
+							<blue>0xffff</blue>
 							<alpha>0xd998</alpha>
 						</default>
 					</option>
@@ -165,9 +165,9 @@
 						<short>Shadow Color</short>
 						<long>Second color of the arrow of the switcher window.</long>
 						<default>
-							<red>0xdc28</red>
-							<green>0xdc28</green>
-							<blue>0xdc28</blue>
+							<red>0x5353</red>
+							<green>0x0000</green>
+							<blue>0x7272</blue>
 							<alpha>0xd998</alpha>
 						</default>
 					</option>
@@ -183,7 +183,7 @@
 				<option name="slide_duration" type="float">
 					<short>Wall Sliding Duration</short>
 					<long>Duration (in s) for wall sliding viewport switching animation</long>
-					<default>0.3</default>
+					<default>1.0</default>
 					<min>0.0</min>
 					<max>5.0</max>
 					<precision>0.05</precision>
@@ -256,12 +256,12 @@
 					<option name="left_window_key" type="key">
 						<short>Move Left With Window</short>
 						<long>Move with window along the wall to the left</long>
-						<default>&lt;Primary&gt;&lt;Alt&gt;&lt;Shift&gt;Left</default>
+						<default></default>
 					</option>
 					<option name="right_window_key" type="key">
 						<short>Move Right With Window</short>
 						<long>Move with window along the wall to the right</long>
-						<default>&lt;Primary&gt;&lt;Alt&gt;&lt;Shift&gt;Right</default>
+						<default></default>
 					</option>
 					<option name="up_window_key" type="key">
 						<short>Move Up With Window</short>
@@ -348,7 +348,7 @@
 				<option name="edgeflip_move" type="bool">
 					<short>Edge Flip Move</short>
 					<long>Flip viewport when moving a window to a screen edge</long>
-					<default>true</default>
+					<default>false</default>
 				</option>
 				<option name="edgeflip_dnd" type="bool">
 					<short>Edge Flip DnD</short>
